addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -214,9 +214,14 @@ def _compute_l10n_latam_available_document_types(self):
     @api.depends('l10n_latam_available_document_type_ids', 'debit_origin_id')
     def _compute_l10n_latam_document_type(self):
         for rec in self.filtered(lambda x: x.state == 'draft'):
-            debit_note = rec.debit_origin_id
             document_types = rec.l10n_latam_available_document_type_ids._origin
-            document_types = document_types.filtered(lambda x: x.internal_type == 'debit_note') if debit_note else document_types
+            invoice_type = rec.move_type
+            if invoice_type in ['out_refund', 'in_refund']:
+                document_types = document_types.filtered(lambda x: x.internal_type not in ['debit_note', 'invoice'])
+            elif invoice_type in ['out_invoice', 'in_invoice']:
+                document_types = document_types.filtered(lambda x: x.internal_type not in ['credit_note'])
+            if rec.debit_origin_id:
+                document_types = document_types.filtered(lambda x: x.internal_type == 'debit_note')
```

Comment:
should not we fix in l10n_ar rather?

Reply:
Code too complicated.
I looked at what is in this file and it works now with something that seems logical to me.

I fixed what I knew was wrong for sure in AR too.

Reply:
I suppose the filtering won't hurt...